fe/fe-core/src/main/java/com/starrocks/alter/LakeRollupJob.java

```diff
@@ -0,0 +1,725 @@
+                long visibleVersion = partition.getVisibleVersion();
+
+                MaterializedIndex rollupIndex = entry.getValue();
+                Map<Long, Long> tabletIdMap = this.physicalPartitionIdToBaseRollupTabletIdMap.get(partitionId);
+                for (Tablet rollupTablet : rollupIndex.getTablets()) {
+                    long rollupTabletId = rollupTablet.getId();
+                    long baseTabletId = tabletIdMap.get(rollupTabletId);
+                    AlterReplicaTask.RollupJobV2Params rollupJobV2Params =
+                            RollupJobV2.analyzeAndCreateRollupJobV2Params(tbl, rollupSchema, whereClause, db.getFullName());
+
+                    ComputeNode computeNode = GlobalStateMgr.getCurrentState().getWarehouseMgr()
+                            .getComputeNodeAssignedToTablet(warehouseId, (LakeTablet) rollupTablet);
+                    if (computeNode == null) {
+                        //todo: fix the error message.
+                        throw new AlterCancelException("No alive backend");
+                    }
+
+                    TabletInvertedIndex invertedIndex = GlobalStateMgr.getCurrentState().getTabletInvertedIndex();
+                    long baseIndexId = invertedIndex.getTabletMeta(baseTabletId).getIndexId();
+                    List<Column> baseColumn = tbl.getIndexMetaByIndexId(baseIndexId).getSchema();
+                    AlterReplicaTask rollupTask = AlterReplicaTask.rollupLakeTablet(
+                            computeNode.getId(), dbId, tableId, partitionId, rollupIndexId, rollupTabletId,
+                            baseTabletId, visibleVersion, jobId,
+                            rollupJobV2Params, baseColumn, watershedTxnId);
+                    rollupBatchTask.addTask(rollupTask);
+                }
+                //partition.setMinRetainVersion(visibleVersion);
+            }
+        }
+
+        sendAgentTask(rollupBatchTask);
+        this.jobState = JobState.RUNNING;
+        span.addEvent("setRunning");
+
+        // DO NOT write edit log here, tasks will be send again if FE restart or master changed.
+        LOG.info("transfer rollup job {} state to {}", jobId, this.jobState);
+    }
+
+    @Override
+    protected void runRunningJob() throws AlterCancelException {
+        Preconditions.checkState(jobState == JobState.RUNNING, jobState);
+
+        // must check if db or table still exist first.
+        // or if table is dropped, the tasks will never be finished,
+        // and the job will be in RUNNING state forever.
+        if (tableHasBeenDropped()) {
+            throw new AlterCancelException("Table or database does not exist");
+        }
+
+        if (!rollupBatchTask.isFinished()) {
+            LOG.info("rollup tasks not finished. job: {}", jobId);
+            List<AgentTask> tasks = rollupBatchTask.getUnfinishedTasks(2000);
+            AgentTask task = tasks.stream().filter(t -> t.getFailedTimes() >= 3).findAny().orElse(null);
```

Comment:
t.isFailed()?

Reply:
this can retry?

Reply:
other type jobs use `t.isFailed() || t.getFailedTimes() >= 3`